app/controllers/search_controller.rb

```diff
@@ -127,8 +130,26 @@ def feed_content
                 elsif params[:class_name] == "User"
                   # No need to check for articles or podcast episodes if we know we only want users
                   user_search
-                else
-                  # if params[:class_name] == PodcastEpisode, Article, or Comment then skip user lookup
+                elsif params[:class_name] == "Article" && params[:search_fields].blank? # homepage
```

Comment:
**suggestion (non-blocking):**
Since this `feed_content` controller action is going to keep growing as we implement each search endpoint, what if we did something like this?

```ruby
def feed_content
  class_name = params[:class_name]&.inquiry
  
  result =
    if class_name.blank?
      # If we are in the main feed and not filtering by type return
      # all articles, podcast episodes, and users
      feed_content_search.concat(user_search)
    elsif class_name.Article? && params[:search_fields].blank?
      # homepage code here
    elsif class_name.Comment? && FeatureFlag.enabled?(:search_2_comments)
      # Search Comments with Postgres FTS
    elsif class_name.Podcast? && FeatureFlag.enabled?(:search_2_podcasts)
      # Search Podcasts (PodcastEpisodes?) with Postgres FTS
    ...
    else
      feed_content_search
    end
      
  render json: {
    result: result,
    display_jobs_banner: SiteConfig.display_jobs_banner,
    jobs_url: SiteConfig.jobs_url
  }
end
```

I'm thinking this will 1) help avoid painful merge conflicts and 2) keep it consistent as we are bouncing PRs back and forth, implementing each search type one at a time.

Reply:
Great idea! :-) I'm going to apply your suggestion to this branch so we can keep working in parallel in case it gets merged before yours. One less headache :D